e2e/multisig-migration.test.ts

```diff
@@ -62,3 +71,229 @@ async function checkForkedNetwork() {
     console.error("Not a forked network:", error);
   }
 }
+
+function generateConcatenatedSignatures(owners: string[]) {
+  const concatenatedSignatures =
+    "0x" +
+    owners
+      .sort((a, b) => a.toLowerCase().localeCompare(b.toLowerCase())) // SORT owners in ascending order
+      .map((owner) => {
+        return "0".repeat(24) + owner.slice(2) + "0".repeat(64) + "01";
+      })
+      .join("");
+
+  return concatenatedSignatures;
+}
+
+export async function multisigExecProviderStatusChangeTransaction(
+  safeAddress: string,
+  lbcAddress: string
+): Promise<boolean> {
+  const safeContract = await ethers.getContractAt("GnosisSafe", safeAddress);
+
+  const EncoderFactory = await ethers.getContractFactory("Encoder");
+  const encoderDeployed = await EncoderFactory.deploy();
+  const encoder = await ethers.getContractAt(
+    "Encoder",
+    await encoderDeployed.getAddress()
+  );
+
+  const callData = await encoder.getCallDataSetProviderStatus(1, false);
```

Comment:
why creating a full contract for this? I think you can just use ethers interface class https://docs.ethers.org/v6/api/abi/#Interface-encodeFunctionData

Reply:
Solved